### PR TITLE
Implement ChiSquare via Gamma

### PR DESCRIPTION
**Pull request by wd60622**
### Motivation for these changes
Simplify by using the gamma distribution directly based on [PyMC PR](https://github.com/pymc-devs/pymc/pull/7007)

### Implementation details
Remove any ChiSquare reference including the rewrites and documentation

### Checklist
+ [x] Explain motivation and implementation 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues, preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes). Note that if they don't, we will [rewrite/rebase/squash the git history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) before merging.
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ChiSquareRV doesn't exist

## New features
- ...

## Bugfixes
- ...

## Documentation
- Removal of ChiSquareRV 

## Maintenance
- ...

### CODE REVIEW

File: pytensor/tensor/random/basic.py [hunk 1]
```diff
@@ -487,56 +487,8 @@ def gamma(shape, rate=None, scale=None, **kwargs):
...
-            returned. Default is `None` in which case a single random variable
-            is returned.
-
-        """
-        return super().__call__(df, size=size, **kwargs)
-
-
-chisquare = ChiSquareRV()
+def chisquare(df, **kwargs):
+    return gamma(shape=df / 2.0, scale=2.0, **kwargs)
```
Comment (ricardoV94) [inline pytensor/tensor/random/basic.py:491]: Let's keep the old docstrings

Comment (wd60622) [reply pytensor/tensor/random/basic.py:520]: Combined the class doc and the `__call__` method's docstrings

File: pytensor/tensor/random/basic.py [hunk 2]
```diff
@@ -487,56 +487,37 @@ def gamma(shape, rate=None, scale=None, **kwargs):
     return _gamma(shape, scale, **kwargs)
 
 
-class ChiSquareRV(RandomVariable):
-    r"""A chi square continuous random variable.
+def chisquare(df, **kwargs):
```
Comment (ricardoV94) [inline pytensor/tensor/random/basic.py:490]:
```suggestion
def chisquare(df, size=None, **kwargs):
```

File: pytensor/tensor/random/basic.py [hunk 3]
```diff
@@ -487,56 +487,37 @@ def gamma(shape, rate=None, scale=None, **kwargs):
...
+    ----------
+    df
+        The number :math:`k` of degrees of freedom. Must be positive.
+    size
+        Sample shape. If the given size is, e.g. `(m, n, k)` then `m * n * k`
+        independent, identically distributed random variables are
+        returned. Default is `None` in which case a single random variable
+        is returned.
+    """
+    return gamma(shape=df / 2.0, scale=2.0, **kwargs)
```
Comment (ricardoV94) [inline pytensor/tensor/random/basic.py:520]:
```suggestion
    return gamma(shape=df / 2.0, scale=2.0, size=size, **kwargs)
```

**Review by ricardoV94 — APPROVED** [review]
Thanks!

**Comment by wd60622** [comment]
> Thanks!

Thanks for the feedback!

**Pull request merged**